### DISCUSSION TITLE
Webhook received code

### DISCUSSION THREAD

**Discussion by mukesh1959** [postalserver/postal#1747] [Help with using Postal]
Hi,

I have installed postal on my server and created webhook event endpoint but i dont have any idea about fetching data from endpoint.

Can anyone help me with the webhook receiver code.

**Comment by willpower232** [comment]
Have a look at https://docs.postalserver.io/developer/webhooks

You don't need to fetch data, Postal will deliver data to you when it receives an email that is routed to your endpoint.

**Comment by mukesh1959** [comment]
Hi ,

I am getting the repsonse on end point when event occur but i dont know how to fetch the data from endpoint url and store it in db.

so if you have any sample code or blog pls help me .

**Reply by willpower232** [reply]
Postal will send you a POST request with a body that contains what you need. What programming language are you using?

**Reply by mukesh1959** [reply]
Hi Team,

I am using PHP

**Reply by willpower232** [reply]
any framework?

**Reply by mukesh1959** [reply]
yes

**Reply by mukesh1959** [reply]
Hi Team,

Any update ?

**Reply by willpower232** [reply]
what framework are you using? they all behave differently so need some more details

**Reply by mukesh1959** [reply]
I am using Laravel framwork (I need an reference code to fetch the data ,even if you have any other framework development code it will be fine)

**Reply by willpower232** [reply]
I think it would be something like this. Then you can look up the message in your database by the ID and token.

```
    public function handle(Request $request)
    {
        $body = $request->getContent();

        $json = json_decode($body);

        if (isset($json['payload']['message'])) {
            $postal_id = $json['payload']['message']['id'];
            $postal_token = $json['payload']['message']['token'];
        } elseif (isset($json['payload']['original_message'])) {
            $postal_id = $json['payload']['original_message']['id'];
            $postal_token = $json['payload']['original_message']['token'];
        }
```
